pkg/crd/parser_integration_test.go

```diff
@@ -132,6 +145,18 @@ var _ = Describe("CRD Generation From Parsing to CustomResourceDefinition", func
 			ExpectWithOffset(1, parser.CustomResourceDefinitions[groupKind]).To(Equal(crd), "type not as expected, check pkg/crd/testdata/README.md for more details.\n\nDiff:\n\n%s", cmp.Diff(parser.CustomResourceDefinitions[groupKind], crd))
 		}
 
+		assertError := func(pkg *loader.Package, kind, errorMsg string) {
+			By(fmt.Sprintf("requesting that the %s CRD be generated", kind))
+			groupKind := schema.GroupKind{Kind: kind, Group: "testdata.kubebuilder.io"}
+			parser.NeedCRDFor(groupKind, nil)
+
+			By(fmt.Sprintf("fixing top level ObjectMeta on the %s CRD", kind))
+			crd.FixTopLevelMetadata(parser.CustomResourceDefinitions[groupKind])
+
+			By("checking that specific errors occurred along the way")
+			Expect(packageErrorsWithMessage(pkg, errorMsg)).To(HaveOccurred())
```

Comment:
An alternative suggestion, you could extract all errors and do something like, minor, depends if you prefer the readability of gomega or not
```
Expect(packageErrors(pkg)).To(ContainElement(MatchError(ContainSubstring(errorMsg))))
```

Reply:
Good suggestion and done :)